client/template_test.go

```diff
@@ -0,0 +1,200 @@
+package client_test
```

Comment:
@eyalatzmon most of this code isn't from the PR, can you sort it out so we'll see a proper diff?

Reply:
merged with master instead the template UT PR